File: THCS/2022_2023/KhanhHoa_NhaTrang_2223/Bai2/chiatao.py
def get_divisors(n):
    """Hàm tìm các ước số của n."""
    divisors = set()
    for i in range(1, int(n**0.5) + 1):
        if n % i == 0:
            divisors.add(i)
            divisors.add(n // i)
    return divisors

def main():
    # Đọc dữ liệu từ file
    with open("chiatao.inp", "r") as f:
        r, g = map(int, f.readline().strip().split())

    # Tìm các ước số của r và g
    divisors_r = get_divisors(r)
    divisors_g = get_divisors(g)

    # Mở file để ghi kết quả
    with open("chiatao.out", "w") as f:
        # Kiểm tra ước số chung và ghi kết quả
        for n in divisors_r:
            if n in divisors_g:
                x = r // n
                y = g // n
                f.write(f"{n} {x} {y}\n")

if __name__ == "__main__":
    main()

File: THCS/2022_2023/KhanhHoa_NhaTrang_2223/Bai2/chiatao.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm> // Thư viện cần thiết cho hàm find

using namespace std;

// Hàm tìm ước số của số nguyên
vector<int> getDivisors(int num) {
    vector<int> divisors;
    for (int i = 1; i * i <= num; i++) {
        if (num % i == 0) {
            divisors.push_back(i);
            if (i != num / i) {
                divisors.push_back(num / i);
            }
        }
    }
    return divisors;
}

int main() {
    ifstream input("chiatao.inp");
    ofstream output("chiatao.out");

    int r, g;
    input >> r >> g;

    // Tìm ước số của r và g
    vector<int> divisorsR = getDivisors(r);
    vector<int> divisorsG = getDivisors(g);

    // Kiểm tra ước số chung và ghi kết quả
    for (int n : divisorsR) {
        if (find(divisorsG.begin(), divisorsG.end(), n) != divisorsG.end()) {
            // Tính số táo đỏ và táo xanh
            int x = r / n;
            int y = g / n;
            output << n << " " << x << " " << y << endl;
        }
    }

    input.close();
    output.close();
    return 0;
}
